### PR TITLE
feat: Global modal for self matchmaking

### PR DESCRIPTION
**Pull request by KSIX2**
When the user clicks the 'simulate' button in dashboard, the dropdown select field of both code and map commit is set to the most recent commit, and the API sends only mode as self, while rest all fields are set as undefined.

When the user clicks 'select' in commit history, the selected commit is loaded into the dropdown select field. In case, the user fails to select one of code or map commit (eg he chooses only code commit and asks for simulation, then the API sends code commit chosen along with map commit field as his recent choice. If there was no recent choice of the user, the API sets map commit as undefined. It works vice versa)

Of course the user can chose any combination of code and map commit from the drop down.

### CODE REVIEW

File: src/components/matchMakingModal/selfMatchModal.module.css [hunk 1]
```diff
@@ -0,0 +1,66 @@
+.selfMatchModal{
```
Comment (CaptainIRS) [inline src/components/matchMakingModal/selfMatchModal.module.css:1]: Change directory and file name to be UpperCamelCase

File: src/components/matchMakingModal/selfMatchModal.tsx [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+  const dispatch = useAppDispatch();
+  const [completeCodeHistory, setCodeHistory] = useState<CodeRevision[]>([]);
+  const [completeMapHistory, setMapHistory] = useState<GameMapRevision[]>([]);
+
+  useEffect(() => {
+    const codeApi = new CodeApi(apiConfig);
+    codeApi
+      .getCodeRevisions()
+      .then(codeResp => {
+        console.log(codeResp);
```
Comment (CaptainIRS) [inline src/components/matchMakingModal/selfMatchModal.tsx:42]: Remove

File: src/components/matchMakingModal/selfMatchModal.tsx [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+      })
+      .catch(codeError => {
+        if (codeError instanceof ApiError) console.log(codeError);
+      });
+
+    const mapApi = new MapApi(apiConfig);
+    mapApi
+      .getMapRevisions()
+      .then(mapResp => {
+        console.log(mapResp);
```
Comment (CaptainIRS) [inline src/components/matchMakingModal/selfMatchModal.tsx:53]: Remove

**Review by CaptainIRS — CHANGES_REQUESTED** [review]

File: src/components/matchMakingModal/selfMatchModal.module.css (hunk 1 above)
Comment (KSIX2) [reply src/components/matchMakingModal/selfMatchModal.module.css:1]: ok

File: src/components/matchMakingModal/selfMatchModal.tsx (hunk 2 above)
Comment (KSIX2) [reply src/components/matchMakingModal/selfMatchModal.tsx]: my bad

File: src/components/matchMakingModal/selfMatchModal.tsx (hunk 3 above)
Comment (KSIX2) [reply src/components/matchMakingModal/selfMatchModal.tsx]: ok

**Review by CaptainIRS — APPROVED** [review]

**Pull request merged**